packages/app/obojobo-repository/server/models/collection.js

```diff
@@ -27,36 +26,104 @@ class RepositoryCollection {
 				{ id }
 			)
 			.then(selectResult => {
-				return new RepositoryCollection(selectResult)
+				return new Collection(selectResult)
 			})
 			.catch(error => {
 				logger.error('fetchById Error', error.message)
 				return Promise.reject(error)
 			})
 	}
 
-	static create({ title = '', user_id }) {
+	static createWithUser(userId) {
 		return db
 			.one(
 				`
-				INSERT INTO repository_collections
-					(title, user_id)
+					INSERT INTO repository_collections
+						(title, group_type, user_id, visibility_type)
+					VALUES
+						('New Collection', 'tag', $[userId], 'private')
+					RETURNING *`,
+				{ userId }
+			)
+			.then(newCollection => {
+				logger.info('user created collection', { userId, collectionId: newCollection.id })
+				return new Collection(newCollection)
+			})
+	}
+
+	static rename(id, newTitle, userId) {
+		return db
+			.one(
+				`UPDATE repository_collections
+				SET title = $[newTitle]
+				WHERE id = $[id]
+				RETURNING *`,
+				{ id, newTitle }
+			)
+			.then(updatedCollection => {
+				const infoObject = {
+					userId,
+					id: updatedCollection.id,
+					title: updatedCollection.title
+				}
+				logger.info('collection renamed', infoObject)
+				return updatedCollection
+			})
+	}
+
+	static addModule(collectionId, draftId, userId) {
+		return db
+			.none(
+				`INSERT INTO repository_map_drafts_to_collections
+					(draft_id, collection_id, user_id)
 				VALUES
-					($[title], $[user_id])
-				RETURNING
-					id,
-					title,
-					user_id as userId,
-					created_at as createdAt`,
-				{
-					title,
-					user_id
+					($[draftId], $[collectionId], $[userId])
+				`,
+				{ collectionId, draftId, userId }
+			)
+			.then(() => {
+				const infoObject = {
+					userId,
+					collectionId,
+					draftId
 				}
+				logger.info('user added module to collection', infoObject)
+			})
+	}
+
+	static removeModule(collectionId, draftId, userId) {
+		return db
+			.none(
+				`DELETE FROM repository_map_drafts_to_collections
+				WHERE
+					draft_id = $[draftId]
+					AND collection_id = $[collectionId]
+				`,
+				{ collectionId, draftId }
+			)
+			.then(() => {
+				const infoObject = {
+					userId,
+					collectionId,
+					draftId
+				}
+				logger.info('user removed module from collection', infoObject)
+			})
+	}
+
+	static delete(id, userId) {
```

Comment:
Similar to the `rename` function, I'm going to go ahead and keep this as-is since if we ended up using it elsewhere, we'd have to implement logging in two places instead of one.